src/snowflake/cli/plugins/snowpark/commands.py

```diff
@@ -378,14 +388,17 @@ def build(
     Builds the Snowpark project as a `.zip` archive that can be used by `deploy` command.
     The archive is built using only the `src` directory specified in the project file.
     """
-
-    assert_project_type("snowpark")
     cli_context = get_cli_context()
+    pd = cli_context.project_definition
+    if not pd.meets_version_requirement("2"):
+        pd = _migrate_v1_snowpark_to_v2(pd)
+
     snowpark_paths = SnowparkPackagePaths.for_snowpark_project(
         project_root=SecurePath(cli_context.project_root),
-        snowpark_project_definition=cli_context.project_definition.snowpark,
+        snowpark_project_definition=pd,
     )
-    log.info("Building package using sources from: %s", snowpark_paths.source.path)
+    log.info("Building package using sources from:")
+    log.info(",".join(str(s) for s in snowpark_paths.sources))
```

Comment:
Instead of that duplication check, move project_definition to signature.

Reply:
I'm not sure if i understand

Reply:
I moved it to a helper function, as it is used in bot `build` and `deploy` commands